configure.ac

```diff
@@ -1,5 +1,13 @@
 dnl ## Process this file with autoconf to produce a configure script.
 
+dnl Include external macro definitions before AC_INIT to remove comments.
```

Comment:
I don't understand this comment. What's the difference between including before & after AC_INIT and how does it affect comments?

Reply:
I'll add better info here.

The generated configure script includes comments starting with `#` if the m4_include() happens after AC_INIT. If it happens before, these comments are cleaned also. But this applies only for the included files.
So, anything that happens before the AC_INIT can have # comments, afterwards, dnl comments are probably better if they need to be removed from the generated configure shell script.

Reply:
even configure.ac itself could start with:

```
# this comment is not present in the generated ./configure

dnl This comment is not present in the generated ./configure
AC_INIT

# this comment is present in the generated ./configure

dnl This comment is never present in the generated ./configure
```

Reply:
Thanks for the explanation. In that case, I think it would make more sense to move the includes after AC_INIT, so we get the same behavior everywhere. Are there any particular comments in the existing files that cause issues? From a quick look it looks like acinclude.m4 uses dnl quite consistently.

Reply:
These ones: https://github.com/php/php-src/blob/master/build/ax_check_compile_flag.m4

Reply:
Is there any documentation on this AC_INIT behavior? I couldn't find anything about this.

Reply:
Not yet in such detail... What is probably meant here by autoconf manual in this context:

https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Coding-Style.html
```
Do not use dnl to introduce comments: most of the comments you are likely to write are either header comments which are not output anyway, or comments that should make their way into configure. There are exceptional cases where you do want to comment special M4 constructs, in which case dnl is right, but keep in mind that it is unlikely.
```

So these so called "header comments" are actually these comments that happen before the AC_INIT call, because that's how the aclocal.m4 file is also included in when building the configure script.

Another answer on this topic on SO:
https://stackoverflow.com/a/3377708

There is another file called acsite.m4 suggested by last changes by Autoconf but this is a system wide file (when installing it in the system's m4 directory). It wouldn't change much in PHP case which doesn't use Automake and aclocal.

Reply:
Difference in generated sizes:
configure with some of these comments in the ax_*.m4 files: 
2483026 bytes
configure with some of these comments in the ax_*.m4 files stripped:
2467862 bytes

so the size difference is very minimal but it's there. I think the macro definitions and inclusions should be done before the AC_INIT call...